air/src/execution_step/air/fold_stream.rs

```diff
@@ -96,22 +102,10 @@ fn execute_iterations<'i>(
             trace_ctx,
         );
         trace_to_exec_err!(trace_ctx.meet_generation_end(fold_id), fold_stream)?;
-
-        result?;
-        if !exec_ctx.subgraph_complete {
-            break;
-        }
+        generation_observer.observe_generation_results(exec_ctx.subgraph_complete, result);
```

Comment:
I wonder why don't we stop at the first error, but return the last one.

Reply:
It's the previous behaviour and after the latest changes it became not deterministic. The issue here in gates, to make them all open deterministically it's crucial to go through all executed states.